webapp/src/pages/Settings.tsx

```diff
@@ -31,48 +82,625 @@ const Settings: React.FC = () => {
+              _.has(partialConfig, "pipelines")
+                ? partialConfig.pipelines
+                : resultingConfig.pipelines,
+              "name"
+            ),
+          });
+        }}
+      />
+    </Box>
+  );
+
+  const switchNullOrDefault = (
+    field: keyof AzimuthConfig,
+    selected: boolean
+  ) => (
     <Checkbox
-      checked={Boolean(resultingConfig[field])}
+      sx={{ paddingTop: 0.5 }}
+      size="small"
+      checked={selected}
       disabled={isError || isFetching}
-      onChange={(_, checked) =>
-        setPartialConfig({ ...partialConfig, [field]: checked ? {} : null })
+      onChange={(...[, checked]) =>
+        setPartialConfig({
+          ...partialConfig,
+          [field]: checked ? data?.[field] ?? CONFIG_SUB_FIELDS[field] : null,
+        })
       }
     />
   );
 
-  return (
-    <Container maxWidth="sm">
-      <FormControl sx={{ width: "100%" }}>
-        <FormGroup>
-          <FormControlLabel
-            control={switchNullOrDefault("behavioral_testing")}
-            label="Perturbation testing"
-          />
+  const displayArgumentsList = (
+    name: string,
+    argEntries: Record<string, string> | Record<string, string>[]
+  ) => (
+    <Box display="flex" flexDirection="column" paddingTop={1}>
+      <Typography variant="caption">{name}</Typography>
+      {Object.entries(argEntries).map(([field, value], index) => (
+        <Box
+          key={index}
+          display="flex"
+          flexDirection="row"
+          justifyContent="flex-start"
+        >
+          <Typography
+            variant="body2"
+            sx={{
+              s: 1,
+              width: "100%",
+            }}
+          >
+            {field}:
+          </Typography>
+          <Tooltip title={value} placement="bottom">
+            <Typography
+              variant="body2"
+              sx={{
+                s: 1,
+                width: "20ch",
+                whiteSpace: "nowrap",
+                overflow: "hidden",
+                textOverflow: "ellipsis",
+              }}
+            >
+              {value}
+            </Typography>
+          </Tooltip>
+        </Box>
+      ))}
+    </Box>
+  );
+
+  const displayReadonlyFields = (label: string, value: string | undefined) => (
+    <TextField
+      sx={{ m: 1, width: "40ch" }}
+      size="small"
+      variant="standard"
+      label={<Typography fontWeight="bold">{label}</Typography>}
+      value={value}
+      disabled={isError || isFetching}
+      InputProps={{
+        readOnly: true,
+        disableUnderline: true,
+        style: { fontSize: 14 },
+      }}
+    />
+  );
+
+  const displayNumberField = (
+    config: keyof AzimuthConfig,
+    field: string,
+    value: number | undefined
+  ) => (
+    <TextField
+      id={field}
+      size="small"
+      label={<Typography fontWeight="bold">{field}</Typography>}
+      type="number"
+      value={value}
+      disabled={!Boolean(resultingConfig[config])}
```

Comment:
No need for `Boolean` if you are inverting with `!`.
```suggestion
      disabled={!resultingConfig[config]}
```